compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -938,32 +938,6 @@ impl<'a, 'tcx> Visitor<'tcx> for BoundVarContext<'a, 'tcx> {
         }
     }
 
-    fn visit_param_bound(&mut self, bound: &'tcx hir::GenericBound<'tcx>) {
-        match bound {
-            hir::GenericBound::LangItemTrait(_, _, hir_id, _) => {
-                // FIXME(jackh726): This is pretty weird. `LangItemTrait` doesn't go
-                // through the regular poly trait ref code, so we don't get another
-                // chance to introduce a binder. For now, I'm keeping the existing logic
-                // of "if there isn't a Binder scope above us, add one", but I
-                // imagine there's a better way to go about this.
```

Comment:
So happy to see this going away.

Reply:
Yes! It's one of the reasons why I took up this PR. Every time I work in the `resolve_bound_vars` code, I'm always worried I'm subtly off-by-one in debruijn indices when considering lang item paths.